trie/trie.go

```diff
@@ -134,6 +134,17 @@ func (t *Trie) NodeIterator(start []byte) (NodeIterator, error) {
 	return newNodeIterator(t, start), nil
 }
 
+// NodeIteratorWithPrefix returns an iterator that returns nodes of the trie with a specific prefix.
+// Iteration starts at the key after the given prefix and stops when leaving the subtree.
+func (t *Trie) NodeIteratorWithPrefix(prefix []byte) (NodeIterator, error) {
+	// Short circuit if the trie is already committed and not usable.
+	if t.committed {
+		return nil, ErrCommitted
+	}
+	// Use NewSubtreeIterator with just a startKey and no stopKey boundary
+	return NewSubtreeIterator(t, prefix, nil), nil
```

Comment:
This makes no sense to me. The iterator will not stop when leaving the subtree no? If stopPrefix is not set, I don't see a functional difference between NoderIteratorWithPrefix and NodeIterator